python/lambda-cloudwatch-dashboard/lambda_cloudwatch_dashboard/lambda_cloudwatch_dashboard_stack.py

```diff
@@ -18,13 +18,14 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
                                     runtime=aws_lambda.Runtime.PYTHON_3_7,
                                     handler="lambda-handler.main",
                                     code=aws_lambda.Code.from_asset("./lambda"))
-        lambda_function.role.add_managed_policy(
-            aws_iam.ManagedPolicy.from_aws_managed_policy_name("service-role/AWSLambdaBasicExecutionRole"))
+        if lambda_function.role is not None:
```

Comment:
When is this potentially `None`?

Reply:
The assert on its own is enough: unless someone explicitly turns off assertions by turning on optimization, it should be fine.